overlord/ifacestate/ifacestate.go

```diff
@@ -332,58 +332,73 @@ func disconnectTasks(st *state.State, conn *interfaces.Connection, flags disconn
 	}
 
 	ts := state.NewTaskSet()
+	var prev *state.Task
+	addTask := func(t *state.Task) {
+		if prev != nil {
+			t.WaitFor(prev)
+		}
+		ts.AddTask(t)
```

Comment:
Small suggestion, this would allow you not to remember to update `prev` after each of the calls to `addTask` below.

```suggestion
		ts.AddTask(t)
		prev = t
```

Reply:
Good idea. Done.